### PR TITLE
Add FV for policy grouping.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Additional FV test for #8098 

## Related issues/PRs

CORE-10015

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

File: felix/fv/policy_grouping_test.go
Comment (aaaaaaaalex) [inline felix/fv/policy_grouping_test.go:81]: Any pre-existing, top-level context that could be used instead of `TODO` that would respond to interrupts?

File: felix/fv/policy_grouping_test.go
Comment (aaaaaaaalex) [inline felix/fv/policy_grouping_test.go:177]: Is it necessary to cleanup test policies here?

Comment (fasaxc) [reply felix/fv/policy_grouping_test.go:177]: No, that's automatic.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

File: felix/fv/policy_grouping_test.go
Comment (fasaxc) [reply felix/fv/policy_grouping_test.go]: Yeah, should really use a timeout; annoying that the test harness doesn't provide one (AFAICT)

File: felix/fv/infrastructure/felix.go
Comment (nelljerram) [inline felix/fv/infrastructure/felix.go:444]: Here also.

File: felix/fv/policy_grouping_test.go
Comment (nelljerram) [inline felix/fv/policy_grouping_test.go:130]: "Group group" ?

File: felix/fv/policy_grouping_test.go
Comment (nelljerram) [inline felix/fv/policy_grouping_test.go:132]: "Group inlined group"?

File: felix/fv/policy_grouping_test.go
Comment (nelljerram) [inline felix/fv/policy_grouping_test.go:134]: Could this also be described as 3 inlined groups?

File: felix/fv/policy_grouping_test.go
Comment (nelljerram) [inline felix/fv/policy_grouping_test.go:186]: Does BPF mode apply to this test?

**Review by nelljerram — APPROVED** [review]
LGTM.  Just some nits for optional consideration.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, squash the commits when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, squash the commits when I merge it, and leave the branch after I've merged it.

**Pull request merged**
